### PR TITLE
rofs-filtered 1.4 (new formula)

### PR DESCRIPTION
**Pull request by fulldecent**
This is my first formula, go easy. There are numerous ugly hacks but it
works for now.

### CODE REVIEW

File: Library/Formula/rofs-filtered.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
+class RofsFiltered < Formula
+  homepage "https://github.com/gburca/rofs-filtered"
+  url "https://github.com/gburca/rofs-filtered/archive/rel-1.4.tar.gz"
+  sha1 "3650f55510f800aa22e64aedf1841c020ce44c08"
+
+  depends_on 'pkg-config' => :build
```
Comment (mikemcquaid) [inline Library/Formula/rofs-filtered.rb]: Use double-quotes everywhere.

File: Library/Formula/rofs-filtered.rb [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  depends_on 'autoconf' => :build
+  depends_on 'automake' => :build
+  depends_on 'libtool' => :build
+  depends_on 'intltool' => :build
+  depends_on 'gettext' => :build
```
Comment (mikemcquaid) [inline Library/Formula/rofs-filtered.rb]: Are you sure this doesn't link against this?

File: Library/Formula/rofs-filtered.rb [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  depends_on :osxfuse
+
+  def install
+    system "./autogen.sh"
+
+    system "echo '#define llistxattr(path, list, size) (listxattr(path, list, size, XATTR_NOFOLLOW))' >> hack.h"
+    system "echo '#define lgetxattr(path, name, value, size) (getxattr(path, name, value, size, 0, XATTR_NOFOLLOW))' >> hack.h"
+    system "echo '#define lsetxattr(path, name, value, size, flags) (setxattr(path, name, value, size, 0, flags | XATTR_NOFOLLOW))' >> hack.h"
+
+    system "echo '#include \"hack.h\"' | cat - rofs-filtered.c > tmp && mv tmp rofs-filtered.c"
```
Comment (mikemcquaid) [inline Library/Formula/rofs-filtered.rb]: Can you please submit this patch to the upstream developers of this project and add a link to the upstream patch submission and explanation of why the patch is needed in a comment in the formula file. Thanks!

File: Library/Formula/rofs-filtered.rb [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+    system "./configure", "--disable-debug",
+                          "--disable-dependency-tracking",
+                          "--disable-silent-rules",
+                          "--prefix=#{prefix}",
+                          "CFLAGS=-I/usr/local/include/osxfuse/fuse",
+                          "LDFLAGS=-L/usr/local/lib"
```
Comment (mikemcquaid) [inline Library/Formula/rofs-filtered.rb]: Use `HOMEBREW_PREFIX` as you can't rely on `/usr/local` being the install location.

File: Library/Formula/rofs-filtered.rb [hunk 5]
```diff
@@ -0,0 +1,36 @@
+class RofsFiltered < Formula
+  homepage "https://github.com/gburca/rofs-filtered"
+  url "https://github.com/gburca/rofs-filtered/archive/rel-1.4.tar.gz"
+  sha1 "3650f55510f800aa22e64aedf1841c020ce44c08"
```
Comment (DomT4) [inline Library/Formula/rofs-filtered.rb]: Should be SHA256.

File: Library/Formula/rofs-filtered.rb (hunk 4 above)
Comment (DomT4) [inline Library/Formula/rofs-filtered.rb]: ` "LDFLAGS=-L/usr/local/lib" ` should already be done by superenv, I think.

File: Library/Formula/rofs-filtered.rb [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+  depends_on 'automake' => :build
+  depends_on 'libtool' => :build
+  depends_on 'intltool' => :build
+  depends_on 'gettext' => :build
+  depends_on :osxfuse
+
+  def install
+    system "./autogen.sh"
+
+    system "echo '#define llistxattr(path, list, size) (listxattr(path, list, size, XATTR_NOFOLLOW))' >> hack.h"
```
Comment (DomT4) [inline Library/Formula/rofs-filtered.rb]: Rather than echo this you could always do ` (buildpath/"hack.h").write <<-EOS.undent `.

**Comment by fulldecent** [comment]
@mikemcquaid  I have updated the quotes and directory. @DomT4 SHA1 is the default which was set up by `brew create` should I really change this?

Thanks for you help. Is this latest better?

**Comment by DomT4** [comment]
> SHA1 is the default which was set up by brew create should I really change this?

If SHA1 is still the checksum produced by ` brew create ` you need to ` brew update ` :).

**Comment by fulldecent** [comment]
@DomT4 I have updated brew and this latest commit uses SHA256

**Comment by DomT4** [comment]
@fulldecent Thanks!

**Comment by DomT4** [comment]
Looks like it failed to find ` fuse `:

```
checking whether to use libfuse... yes
checking fuse.h usability... no
checking fuse.h presence... no
checking for fuse.h... no
checking for fuse_mount in -lfuse... no
configure: error: can not use the `fuse' library!
      * fuse library `libfuse' not found or
      * fuse header `fuse.h' not found
      Please install the library and its header files.
      If you are working with a package based distribution look for
      package with a name ending with `-dev' (the developement packages).
      If you have installed the library and the header files
      but they were not found use `--with-libfuse=DIR'
      when the library is installed in DIR/lib
      and the header files are installed in DIR/include.
```

File: Library/Formula/rofs-filtered.rb (hunk 3 above)
Comment (gburca) [inline Library/Formula/rofs-filtered.rb]: Patch is now in master (https://github.com/gburca/rofs-filtered/commit/151f131482f3daf0fa4f3f550cca01fcb9f972f9)

Comment (mikemcquaid) [inline Library/Formula/rofs-filtered.rb]: Needs a packaged release but can use this patch as a `patch do` with https://github.com/gburca/rofs-filtered/commit/151f13.patch

**Comment by fulldecent** [comment]
Discussion moved to https://github.com/Homebrew/homebrew/pull/38436

**Pull request closed without merging**